### PR TITLE
PR: Fix Fileswitcher for 'EditorStack' instances #2

### LINKED ISSUE #10684: EditorStack file and symbol switcher not working #2
### What steps reproduce the problem?

Click on `File switcher...` or ` Find symbol in file...` of an editorstack menu. 

image
 

### Paste Traceback/Error Below (if applicable)

```python-traceback
Traceback (most recent call last):
  File "C:\Users\User\spyder\spyder\plugins\editor\widgets\editor.py", line 897, in open_switcher_dlg
    section=self.get_plugin_title())
TypeError: __init__() missing 1 required positional argument: 'get_editorstack'
```

## Versions

* Spyder 4.0.0.dev0 (Commit: 327c67a64) 
* Python 3.7.3 64-bit | Qt 5.9.6 | PyQt5 5.9.2 | Windows 10

### PR DESCRIPTION
**Pull request by jnsebgosselin**
## Description of Changes

* [x] Wrote at least one-line docstrings (for any new functions)
* [x] Added unit test(s) covering the changes (if testable)

### Issue(s) Resolved
Fixes #10684

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Jean-Sébastien Gosselin

### CODE REVIEW

**Comment by jnsebgosselin** [comment]
I guess a unit test should be added too this is the second time the switcher is broken for the editorstack.

**Comment by goanpeca** [comment]
@jnsebgosselin yes, please add a test on the `spyder/app/test_mainwindow.py`

There are many examples on that file you could use like https://github.com/spyder-ide/spyder/blob/master/spyder/app/tests/test_mainwindow.py#L2110

File: spyder/app/tests/test_mainwindow.py [hunk 1]
```diff
@@ -1815,6 +1815,57 @@ def example_def_2():
...
+    for i in range(3):
+        file = tmpdir.join('test_file{}.py'.format(i))
+        file.write("some_content_for_test_file_{}".format(i))
+        main_window.editor.load(str(file))
+
+    # Test that the file switcher opens as expected from the editorstack.
+    editorstack = main_window.editor.get_current_editorstack()
+    assert editorstack.switcher_dlg is None
+    editorstack.open_switcher_dlg()
+    assert editorstack.switcher_dlg
```
Comment (goanpeca) [inline spyder/app/tests/test_mainwindow.py]: I think we should probably still assert that the dialog is visible
`editorstack.switcher_dlg.isVisible()` ?

What do you think?

File: spyder/app/tests/test_mainwindow.py [hunk 2]
```diff
@@ -1815,6 +1815,57 @@ def example_def_2():
...
+               def example_def_2():
+                   pass
+               ''')
+    main_window.editor.load(str(file))
+
+    # Test that the symbol finder opens as expected from the editorstack.
+    editorstack = main_window.editor.get_current_editorstack()
+    assert editorstack.switcher_dlg is None
+    editorstack.open_symbolfinder_dlg()
+    assert editorstack.switcher_dlg
```
Comment (goanpeca) [inline spyder/app/tests/test_mainwindow.py]: I think we should probably still assert that the dialog is visible
`editorstack.switcher_dlg.isVisible()` ?

What do you think?

**Comment by ccordoba12** [comment]
Note: According to Azure all builds passed, so the error here was reported incorrectly.

**Pull request merged**